### ISSUE TITLE
edit txt in vst and pop menu

### ISSUE THREAD

**Issue by onukore** [kmatheussen/radium#648]
hi,
- i use redux to edit my sample in radium, and impossible to enter text to save sample (and import it in radium sampler) the cursor is in text box but i can't wirte text, and that for all plugin in want to wirte text in vst gui (in redux save sample is under sample browser)
in old version we can enter text now is impossible
- in vst when i open pop up menu this one is always beyond the vst gui ( this bug is here since vst gui always on top) in some some  vst right-clic open menu (redux, tunefish ect ...) i can see only part of menu, other part as hidden by vst gui

by cu
ps: we need sample editor to basic edit (cut,copy etc... take redux to example) 
ps2: audio recording is very good, an rec botton on track header for easyer multitrack rec ,was very useful if is possible
ps3 : now we can rec audio , what about sample seeking (start sample any where in block even if no note to triggers) , wery useful  to work on rang loop mode with big sample as note trigged in rang

**Comment by kmatheussen** [comment]
(Please make several issues, one for each topic. The discussion threads becomes hard to follow and there's often misunderstandings about which topic is being talked about in the comments, and it's hard to find a specific issue from the main list of issues this way.)

> in vst when i open pop up menu this one is always beyond the vst gui

Are you talking about a popup menu inside a native VST gui, or are you talking about a popup menu in Radium?

**Comment by onukore** [comment]
sorry !
yes inside native VST gui

**Comment by kmatheussen** [comment]
> yes inside native VST gui

Sounds very much like a bug in the VST plugin. Does the same thing happen for this vst plugin in Aodix? (which I think also forces native vst gui to front)

**Comment by onukore** [comment]
no that not happen in aodix or tracktion (with same plugin), that happen since always on top plugin ,

**Comment by kmatheussen** [comment]
I forgot this one too.

**Comment by kmatheussen** [comment]
Hi, 

I've thought a lot about it, and this must be fixed in the VST plugins,
not in Radium. If there are VST hosts that are able to show these
combo boxes in always-on-top mode, they are doing black magic,
or they will fail in some situations. I'm pretty sure it's impossible
to fix this on the host side. The sane way to fix this is for the VST
plugins to make sure their sub windows are always shown in
front of the main window.

But as a workaround, I've added option to turn off always-on-top mode.
Maybe I'll add some semi-magic code in the future, namely
to automatically raise the vst plugin windows if the radium window
is shown in front, but that won't always work, and it's probably a
lot of work to implement as well.

**Comment by onukore** [comment]
hi
aodix : 
![capture d ecran 2016-08-29 16 07 54](https://cloud.githubusercontent.com/assets/13727495/18054307/96bba5a4-6e03-11e6-9a3d-1207f9752c89.png)

renoise : 

![capture d ecran 2016-08-29 16 11 17](https://cloud.githubusercontent.com/assets/13727495/18054427/14c7d4a4-6e04-11e6-8d4f-3cd4274d3b25.png)

no black magic ;)

**Comment by kmatheussen** [comment]
It's not impossible to make it seem like it works, but either it's not a perfect solution or it's some kind of black magic. The correct solution is for the plugin makers to fix this.

**Comment by kmatheussen** [comment]
I have a seemingly working solution running now, but it's not pretty.

**Comment by onukore** [comment]
I don't think it's a plugin bug, because that work fine in all other host i have tryed ( reaper, live, tracktion, buzz ect...) if it's not a plugin bug, it's a radium bug..

**Comment by kmatheussen** [comment]
It's definitely a bug in the plugin when a sub window pops up behind the main window. There's long discussions about this problem on the juce forum, where this happened in traction and other hosts. But I guess it's more natural for the hosts to work around these corner cases, since there are so fewer hosts.

**Comment by kmatheussen** [comment]
After digging in the windows programming documentation, I finally found a single command that magically fixed everything (```SetWindowLongPtr(child_hwnd, -8, (LONG_PTR)parent_hwnd)```).

**Comment by onukore** [comment]
Nice to ear, good work .!

**Comment by kmatheussen** [comment]
> ps2: audio recording is very good, an rec botton on track header for easyer multitrack rec ,was very useful if is possible

Can you explain this idea in more detail?

**Comment by Lykdraft** [comment]
I guess what he means is maybe:

a) that you have a "classic" record button to enable all samplers at once into record mode so that you can record into multiple samplers at once with a press of a button.
(don't know if that makes a lot of sense though, as this would overwrite any samplers that already do something else...)

or b) add a record button to every header of every single track in the tracklist,
so that you can click the record button faster without the need to go to the instrument widget
(with the ability to switch tracks so friggin fast with keyboard-shortcuts, I don't know if a proposal like this isn't just pure lazyness.. :) )
